st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -130,13 +130,16 @@ def pre_run(self):
             raise runnerexceptions.ActionRunnerPreRunError(message)
 
     def run(self, action_parameters):
-
         result = {'tasks': []}  # holds final result we store
         context_result = {}  # holds result which is used for the template context purposes
         top_level_error = None  # stores a reference to a top level error
         fail = True
         action_node = None
 
+        # Add action parameters to the context result so they can be accessed
+        # inside "publish"
+        context_result['parameters'] = action_parameters
```

Comment:
To clarify it - inside the `params" section, action parameters are accessible in a root scope (there is no prefix), so we could also drop the prefix all together in the `publish` section.

E.g.

```yaml
---
  chain:
    -
      name: "create_temporary_directory"
      ref: "core.remote_sudo"
      params:
        hosts: "{{ ip_address }}"  -> no prefix here
        cmd: "mktemp -d"
      publish:
        temp_dir: "{{ create_temporary_directory[parameters.ip_address].stdout }}" -> we probably shouldn't have a prefix here for consistency reasons
      on-success: "retrieve_logged_users"
```